pkg/machines/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -366,6 +428,53 @@ class CreateVM extends React.Component {
                            onChange={this.onChangedEventChecked.bind(this, 'startVm')} />
                     {_("Immediately Start VM")}
                 </label>
+
+                <hr />
+
+                { this.props.providerName == 'LibvirtDBus' &&
+                this.state.sourceType != EXISTING_DISK_IMAGE_SOURCE &&
+                    <React.Fragment>
+                        <label className="checkbox-inline">
+                            <input id="choose-storage" type="checkbox"
+                                   checked={this.state.chooseStorage}
+                                   onChange={this.onChangedEventChecked.bind(this, 'chooseStorage')} />
+                            {_("Choose Storage")}
+                        </label>
+
+                        { this.state.chooseStorage &&
+                            <React.Fragment>
+                                <label className="control-label" htmlFor="storage-pool">
+                                    {_("Storage Pool")}
+                                </label>
+                                <Select.Select id="storage-pool-select"
+                                               initial={this.state.storagePool}
+                                               onChange={this.onChangedValue.bind(this, 'storagePool')}>
+                                    { this.props.storagePools.map(pool => {
+                                        if (pool.connectionName === this.state.connectionName)
```

Comment:
This filter should go away when you do the filtering at the outermost component for CreateVm dialog.